controlplane/internal/controller/openshiftassistedcontrolplane_controller.go

```diff
@@ -246,6 +247,7 @@ func (r *OpenshiftAssistedControlPlaneReconciler) upgradeWorkloadCluster(ctx con
 	oacp.Status.DistributionVersion, err = upgrader.GetCurrentVersion(ctx)
 	if err != nil {
 		log.V(logutil.WarningLevel).Info("failed to get OpenShift version from ClusterVersion", "error", err.Error())
+		return ctrl.Result{}, err
```

Comment:
Why returning here? Even without current version we should be able to perform an upgrade